src/main/java/frc/team2412/robot/subsystem/ShooterVisionSubsystem.java

```diff
@@ -1,24 +1,39 @@
 package frc.team2412.robot.subsystem;
 
 import static frc.team2412.robot.subsystem.ShooterVisionSubsystem.ShooterVisionConstants.*;
+
+import org.frcteam2910.common.drivers.Gyroscope;
+import org.frcteam2910.common.math.RigidTransform2;
+import org.frcteam2910.common.math.Rotation2;
+import org.frcteam2910.common.math.Vector2;
+
 import edu.wpi.first.networktables.NetworkTable;
 import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ShooterVisionSubsystem extends SubsystemBase {
-
     public static class ShooterVisionConstants {
-
-        public static double LIMELIGHT_ANGLE_OFFSET = 0;
-        public static double LIMELIGHT_HEIGHT_OFFSET = 0;
-        public static double RIM_HEIGHT = 104; // 8ft8in
-        public static double HEIGHT_TO_RIM = RIM_HEIGHT - LIMELIGHT_HEIGHT_OFFSET;
+        // Angles are in degrees
+        public static final double DEFAULT_PITCH = 45;
+        public static final double LIMELIGHT_ANGLE_OFFSET = 0;
+        // Heights are in inches
+        public static final double LIMELIGHT_HEIGHT_OFFSET = 0;
+        public static final double RIM_HEIGHT = 8 * 12 + 8;
+        public static final double HEIGHT_TO_RIM = RIM_HEIGHT - LIMELIGHT_HEIGHT_OFFSET;
+        // Rotation2 can be specified by degrees or radians
+        public static final Rotation2 STARTING_ROBOT_ROTATION = Rotation2.ZERO; // Placeholder
     }
 
     public NetworkTable limelight;
 
-    public ShooterVisionSubsystem() {
-        limelight = NetworkTableInstance.getDefault().getTable("limelight");
+    private final Gyroscope gyro;
+
+    private final Rotation2 gyroAdjustmentAngle;
+
+    public ShooterVisionSubsystem(Gyroscope gyro) {
+        this.limelight = NetworkTableInstance.getDefault().getTable("limelight");
+        this.gyro = gyro;
+        this.gyroAdjustmentAngle = STARTING_ROBOT_ROTATION.rotateBy(gyro.getUnadjustedAngle().inverse());
```

Comment:
What is the `inverse()` call used for?

Reply:
We know that the `gyro.getUnadjustedAngle()` should be `STARTING_ROBOT_ROTATION`, so to get how much to add to the gyro's angle measurements, we get can add `STARTING_ROBOT_ROTATION - gyro.getUnadjustedAngle()`.